File: src/BusinessLayer/Preset.cpp
#include "Preset.h"

Preset::Preset(int id, std::string presetName, std::vector<int> plateList)
{
    m_ID = id;
    m_PresetName = presetName;
    m_PlateList = plateList;
}

int Preset::GetID()
{
    return m_ID;
}

std::string Preset::GetPresetName()
{
    return m_PresetName;
}

std::vector<int> Preset::GetPlatelist()
{
    return m_PlateList;
}

File: src/UI/CalibrationPlateChangerClient/CalibrationPlateChangerClient/Preset.cs
using System.Collections.Generic;

namespace CalibrationPlateChangerClient {
    public class Preset {
        private int m_ID;
        private string m_Name;
        private List<Plate> m_PlateList;

        public Preset(int id, string name, List<Plate> platelist) {
            m_ID = id;
            m_Name = name;
            m_PlateList = platelist;
        }

        public int GetID() {
            return m_ID;
        }

        public string GetName() {
            return m_Name;
        }

        public List<Plate> GetPlates() {
            return m_PlateList;
        }
    }
}
